ledger/participant-integration-api/src/main/scala/platform/store/utils/ConcurrencyLimiter.scala

```diff
@@ -24,6 +24,7 @@ class QueueBasedConcurrencyLimiter(
   private val waiting = mutable.Queue[Task]()
   private var running: Int = 0
 
+  // is this blocked on a monitor before Future is returned to the consumer?
```

Comment:
@mziolekda What about this? Is it "expensive computation"?

Reply:
actually I have just realised it is not, blocking happens within an execution context